### PR TITLE
[core] Add non historical variables and RMS method for time average process

### PR DESCRIPTION
**Pull request by riccardotosi**
Dear all,

this PR adds

-  support for non historical nodal and elemental variables. Therefore, the averaged variables list must now be specified,

- RMS method.

Tests are updated.

### CODE REVIEW

File: kratos/processes/time_averaging_process.cpp [hunk 1]
```diff
@@ -77,32 +90,41 @@ int TimeAveragingProcess::Check()
...
+        {
+            const std::string& variable_name = mVariableNamesList[counter];
+            const std::string& averaged_variable_name = mAveragedVariableNamesList[counter];
+
+            if (!(
+                (KratosComponents<Variable<double>>::Has(variable_name) && KratosComponents<Variable<double>>::Has(averaged_variable_name)) ||
+                (KratosComponents<Variable<array_1d<double, 3>>>::Has(variable_name) && KratosComponents<Variable<array_1d<double, 3>>>::Has(averaged_variable_name))
+                ))
+                {
+                    KRATOS_ERROR << "List of input variables" << mVariableNamesList << " and list of averaged variables" << mAveragedVariableNamesList << " has different variable types. Please provide arrays of same length and corresponding variables of the same type.\n";
```
Comment (sunethwarna) [inline kratos/processes/time_averaging_process.cpp]:
```suggestion
                    KRATOS_ERROR << variable_name << " type and its corresponding averaging variable " << averaged_variable_name << " has different types. Please provide corresponding variables of the same type.\n";
```

**Comment by sunethwarna** [comment]
I removed if blocks from nodal, and elemental loops. Would this implementation ok with you? @riccardotosi

**Comment by riccardotosi** [comment]
Yes it is ok for me!

**Review by sunethwarna — APPROVED** [review]

**Review by rubenzorrilla — APPROVED** [review]
@riccardotosi what I told you would require partial template specialization to be properly done. Unfortunately, this feature can't be used yet.

**Pull request merged**